src/DropDownMenu/DropDownMenu.js

```diff
@@ -274,15 +294,28 @@ class DropDownMenu extends Component {
 
   handleItemTouchTap = (event, child, index) => {
```

Comment:
> The logic duplication is mostly in Menu. If breaking changes could be made then I think Menu's onItemTouchTab should be removed in favor of an updated onChange with a richer set of parameters.

I agree but I would definitely avoid introducing breaking changes on the `master`. We have the `next` branch to move forward. That current logic sounds like a good tradeoff for now.